src/IK/include/BipedalLocomotion/IK/DistanceTask.h

```diff
@@ -0,0 +1,149 @@
+                                                                              the robot velocity
+                                                                              (base + joint) */
+
+    static constexpr std::size_t m_spatialVelocitySize{6}; /**< Size of the spatial velocity vector.
+                                                            */
+    static constexpr std::size_t m_DoFs{1}; /**< DoFs associated to the task */
+
+    bool m_isInitialized{false}; /**< True if the task has been initialized. */
+    bool m_isValid{false}; /**< True if the task is valid. */
+
+    std::shared_ptr<iDynTree::KinDynComputations> m_kinDyn; /**< Pointer to a KinDynComputations
+                                                               object */
+
+    Eigen::MatrixXd m_jacobian, m_relativeJacobian; /**< Internal buffers to store the jacobian. */
+    Eigen::Vector3d m_framePosition; /**< Internal buffer to store the frame position. */
+    double m_kp; /**< Controller gain. */
+    double m_desiredDistance{0.0}; /**< Desired distance. */
+
+    std::string m_baseName; /**< Base frame name. */
+    std::string m_targetFrameName; /**< Target frame name. */
+    iDynTree::FrameIndex m_baseIndex, m_targetFrameIndex; /**< Base and target frame name indexes.
+                                                           */
+
+    double m_computedDistance{0.0}; /**< Computed distance. */
+
+public:
+    // clang-format off
+    /**
+     * Initialize the task.
+     * @param paramHandler pointer to the parameters handler.
+     * @note the following parameters are required by the class
+     * |           Parameter Name           |              Type            |                                                        Description                                                                       | Mandatory |
+     * |:----------------------------------:|:----------------------------:|:----------------------------------------------------------------------------------------------------------------------------------------:|:---------:|
+     * |   `robot_velocity_variable_name`   |            `string`          |                            Name of the variable contained in `VariablesHandler` describing the robot velocity                            |    Yes    |
+     * |                `kp`                |            `double`          |                                               Gain of the distance controller                                                            |    Yes    |
+     * |         `target_frame_name`        |            `string`          |                                       Name of the frame of which computing the distance                                                  |    Yes    |
+     * |          `base_frame_name`         |            `string`          |          Name of the frame with respect to which computing the distance. If empty, the world frame will be used (Default = "")           |    No     |
```

Comment:
I don't think whether `base_frame_name` is misleading. At first glance, I was thinking to the `root_link`. What do you think about `reference_frame`?

Reply:
Done in https://github.com/ami-iit/bipedal-locomotion-framework/pull/717/commits/5bda80d79a29a6e418e0d14d2509232f39dbacce. @EhsanRanjbari you may want to check this modification